build.gradle.kts

```diff
@@ -0,0 +1,340 @@
+    apply(plugin = "com.android.${if (name == "app" || name == "proguard-tests") {
+        "application"
+    } else {
+        "library"
+    }}")
+
+    configure<BaseExtension> {
+        compileSdkVersion(Config.SdkVersions.compile)
+
+        defaultConfig {
+            minSdkVersion(Config.SdkVersions.min)
+            targetSdkVersion(Config.SdkVersions.target)
+
+            versionName = Config.version
+            versionCode = 1
+
+            resourcePrefix("fui_")
+            vectorDrawables.useSupportLibrary = true
+        }
+
+        lintOptions {
+            disable(
+                    "ObsoleteLintCustomCheck", // TODO ButterKnife will fix this in v9.0
+                    "IconExpectedSize",
+                    "InvalidPackage", // Firestore uses GRPC which makes lint mad
+                    "NewerVersionAvailable", "GradleDependency", // For reproducible builds
+                    "SelectableText", "SyntheticAccessor" // We almost never care about this
+            )
+
+            isCheckAllWarnings = true
+            isWarningsAsErrors = true
+            isAbortOnError = true
+
+            baselineFile = file("$configDir/lint-baseline.xml")
+            htmlOutput = file("$reportsDir/lint-results.html")
+            xmlOutput = file("$reportsDir/lint-results.xml")
+        }
+    }
+}
+
+fun Project.configureQuality() {
+    apply(plugin = "checkstyle")
+
+    configure<CheckstyleExtension> { toolVersion = "8.10.1" }
+    check { dependsOn("checkstyle") }
+
+    task<Checkstyle>("checkstyle") {
+        configFile = file("$configDir/checkstyle.xml")
+        source("src")
+        include("**/*.java")
+        exclude("**/gen/**")
+        classpath = files()
+    }
+}
+
+fun Project.setupPublishing(isLibrary: Boolean) {
+    val publicationName = "${if (isLibrary) "monolith" else name}Library"
+    val artifactName = if (isLibrary) "firebase-ui" else "firebase-ui-${project.name}"
+
+    val sourcesJar = task<Jar>("sourcesJar") {
+        classifier = "sources"
+        from(project.the<BaseExtension>().sourceSets["main"].java.srcDirs)
+    }
+
+    val javadoc = task<Javadoc>("javadoc") {
+        setSource(project.the<BaseExtension>().sourceSets["main"].java.srcDirs)
+        classpath += configurations["compile"]
+        classpath += project.files(project.the<BaseExtension>().bootClasspath)
+    }
+
+    val javadocJar = task<Jar>("javadocJar") {
+        dependsOn(javadoc)
+        classifier = "javadoc"
+        from(javadoc.destinationDir)
+    }
+
+    artifacts.add("archives", javadocJar)
+    artifacts.add("archives", sourcesJar)
+
+    tasks.whenTaskAdded {
+        if (name.contains("publish") && name.contains("publication", true)) {
+            dependsOn("assembleRelease")
+        }
+    }
+
+    afterEvaluate {
+        if (isLibrary) {
+            task("testAll") {
+                dependsOn(*Config.submodules.map {
+                    ":$it:testDebugUnitTest"
```

Comment:
Above we always do `${foo}` even when `$foo` would work and I think that's a good clarity standard, let's do that for all these `$it`s

Reply:
I think that just takes some getting used to. Also, it goes directly against the official style guide: https://kotlinlang.org/docs/reference/coding-conventions.html#string-templates.

If you have a strong opinion on this I won't mind, but I _am_ trying to wave an official document at you to scare you off. 😆